src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/DomTests.cs

```diff
@@ -239,5 +249,548 @@ public static void SerializeToNode_WithEscaping()
+            string json = Encoding.UTF8.GetString(bytes);
+            Assert.Contains("Hello", json);
+        }
+
+        [Fact]
+        public static void SerializeToStream_WithJsonTypeInfo_NonGeneric()
+        {
+            MyPoco obj = MyPoco.Create();
+            
+            JsonSerializerOptions options = new() { TypeInfoResolver = new DefaultJsonTypeInfoResolver() };
+            JsonTypeInfo typeInfo = options.GetTypeInfo(typeof(MyPoco));
+            
+            using MemoryStream stream = new();
+            JsonSerializer.Serialize(stream, obj, typeInfo);
+            
+            stream.Position = 0;
+            string json = new StreamReader(stream).ReadToEnd();
+            Assert.Contains("Hello", json);
+        }
+
+        [Fact]
+        public static void SerializeToUtf8JsonWriter_WithJsonTypeInfo_NonGeneric()
+        {
+            MyPoco obj = MyPoco.Create();
+            
+            JsonSerializerOptions options = new() { TypeInfoResolver = new DefaultJsonTypeInfoResolver() };
+            JsonTypeInfo typeInfo = options.GetTypeInfo(typeof(MyPoco));
+            
+            using MemoryStream stream = new();
+            using (Utf8JsonWriter writer = new(stream))
+            {
+                JsonSerializer.Serialize(writer, obj, typeInfo);
+            }
+            
+            string json = Encoding.UTF8.GetString(stream.ToArray());
+            Assert.Contains("Hello", json);
+        }
+
+        [Fact]
+        public static void DeserializeFromUtf8JsonReader_WithJsonTypeInfo()
+        {
+            byte[] utf8Json = Encoding.UTF8.GetBytes(Json);
+            Utf8JsonReader reader = new(utf8Json);
+            
+            JsonSerializerOptions options = new() { TypeInfoResolver = new DefaultJsonTypeInfoResolver() };
+            JsonTypeInfo<MyPoco> typeInfo = (JsonTypeInfo<MyPoco>)options.GetTypeInfo(typeof(MyPoco));
+            
+            MyPoco obj = JsonSerializer.Deserialize(ref reader, typeInfo);
+            obj.Verify();
+        }
+
+        [Fact]
+        public static void DeserializeFromUtf8JsonReader_WithJsonTypeInfo_NonGeneric()
+        {
+            byte[] utf8Json = Encoding.UTF8.GetBytes(Json);
+            Utf8JsonReader reader = new(utf8Json);
+            
+            JsonSerializerOptions options = new() { TypeInfoResolver = new DefaultJsonTypeInfoResolver() };
+            JsonTypeInfo typeInfo = options.GetTypeInfo(typeof(MyPoco));
+            
+            object obj = JsonSerializer.Deserialize(ref reader, typeInfo);
+            Assert.IsType<MyPoco>(obj);
+            ((MyPoco)obj).Verify();
+        }
+
+        [Fact]
+        public static void DeserializeFromStream_WithJsonTypeInfo()
+        {
+            using MemoryStream stream = new(Encoding.UTF8.GetBytes(Json));
+            
+            JsonSerializerOptions options = new() { TypeInfoResolver = new DefaultJsonTypeInfoResolver() };
```

Comment:
This is equivalent to using `JsonSerializerOptions.Default` and is very wasteful to recreate caches for every unit test.